src/main/java/com/vackosar/gitflowincrementalbuild/boundary/Configuration.java

```diff
@@ -174,12 +177,25 @@ public static boolean isMakeBehaviourActive(String expectedMakeBehavior, MavenSe
         return expectedMakeBehavior.equals(actualMakeBehavior) || MavenExecutionRequest.REACTOR_MAKE_BOTH.equals(actualMakeBehavior);
     }
 
-    private static Properties[] getProperties(MavenSession session, Logger logger) {
-        MavenProject mavenProject = Optional.ofNullable(session.getTopLevelProject()).orElseGet(session::getCurrentProject);
-        if (mavenProject != null) {
-            return new Properties[] { mavenProject.getProperties(), getPluginProperties(mavenProject) };
+    private static MavenProject findCurrentProject(MavenSession session) {
+        // MavenSession.getCurrentProject() does not return the correct value in some cases,
+        // see: https://issues.apache.org/jira/browse/MNG-6979
+        MavenProject currentProject = session.getCurrentProject();
+        if (currentProject == null || !currentProject.isExecutionRoot()) {
+            currentProject = session.getProjects().stream()
+                    .filter(MavenProject::isExecutionRoot)
+                    .findAny()
```

Comment:
`findFirst()` may be more adequate since the project is among the initial values of the list

Reply:
Scratch that, I think it doesn't matter in this case

Reply:
I was a bit on the fence here. There should be only one project having this flag set.
And you know what they say: Use `findFirst` if you really care about the order, use the more efficient `findAny` otherwise.
I am actually rather sure that it makes no difference at all for a sequential stream based on a vanilla list (should start from the beginning regardless) but I might be wrong.